### PR TITLE
Add support for Mac OS X 10.4 on 32-bit PowerPC

### PR DESCRIPTION
**Pull request by turbolent**

### CODE REVIEW

File: lib/pure/os.nim [hunk 1]
```diff
@@ -296,6 +296,10 @@ proc setCurrentDir*(newDir: string) {.inline, tags: [].} =
   else:
     if chdir(newDir) != 0'i32: raiseOSError(osLastError())
 
+when defined(posix):
+   var
+     pathMax {.importc: "PATH_MAX", header: "<stdlib.h>".}: cint
```
Comment (Varriount) [inline lib/pure/os.nim]: Isn't this already declared elsewhere?

Comment (turbolent) [reply lib/pure/os.nim]: Is it? I couldn't find any reference to it. Do you know where?

I just brought this back from an earlier commit which was reverted later, but maybe now this definition is not necessary anymore

File: tools/niminst/buildsh.tmpl [hunk 2]
```diff
@@ -100,8 +100,14 @@ case $uos in
     ;;
   *darwin* )
     myos="macosx"
-    CC="clang"
-    LINKER="clang"
+    case $HOSTTYPE in
```
Comment (dom96) [inline tools/niminst/buildsh.tmpl]: Merge this with the ``if`` statement below.

**Comment by Araq** [comment]
No update after 6 months, closing.

**Pull request closed without merging**